pkg/search/postgres/common.go

```diff
@@ -846,3 +845,27 @@ func redactedQueryData(query *query) interface{} {
 	// Otherwise, allow the logging
 	return query.Data
 }
+
+func scanRows[T any, PT unmarshaler[T]](rows pgx.Rows) ([]*T, error) {
+	var results []*T
+	for rows.Next() {
+		msg, err := unmarshal[T, PT](rows)
+		if err != nil {
+			return nil, err
+		}
+		results = append(results, (*T)(msg))
+	}
+	return results, nil
+}
+
+func unmarshal[T any, PT unmarshaler[T]](row pgx.Row) (PT, error) {
+	var data []byte
+	if err := row.Scan(&data); err != nil {
+		return nil, errors.Wrap(err, "scanning row")
+	}
+	msg := PT(new(T))
+	if err := msg.Unmarshal(data); err != nil {
+		return nil, errors.Wrapf(err, "unmarshaling %T", msg)
+	}
+	return msg, nil
+}
```

Comment:
Super inconsequential, but I think this should rather return a `*T`. You can even do with only a single explicit cast:
```suggestion
func unmarshal[T any, PT unmarshaler[T]](row pgx.Row) (*T, error) {
	var data []byte
	if err := row.Scan(&data); err != nil {
		return nil, errors.Wrap(err, "scanning row")
	}
	msg := new(T)
	if err := PT(msg).Unmarshal(data); err != nil {
		return nil, errors.Wrapf(err, "unmarshaling %T", msg)
	}
	return msg, nil
}
```